app/assets/javascripts/initializers/initializeArticleReactions.js

```diff
@@ -132,39 +81,98 @@ function reactToArticle(articleId, reaction) {
     });
 }
 
-
 function setCollectionFunctionality() {
   if (document.getElementById('collection-link-inbetween')) {
-    var inbetweenLinks = document.getElementsByClassName('collection-link-inbetween');
-    var inbetweenLinksLength = inbetweenLinks.length
-    for (var i = 0; i < inbetweenLinks.length; i++) {
-      inbetweenLinks[i].onclick = function(e) {
+    var inbetweenLinks = document.getElementsByClassName(
+      'collection-link-inbetween',
+    );
+    var inbetweenLinksLength = inbetweenLinks.length;
+    for (var i = 0; i < inbetweenLinks.length; i += 1) {
+      inbetweenLinks[i].onclick = e => {
         e.preventDefault();
         var els = document.getElementsByClassName('collection-link-hidden');
-        var elsLength = els.length
-        for (var i = 0; i < elsLength; i++) {
+        var elsLength = els.length;
+        for (var j = 0; j < elsLength; j += 1) {
           els[0].classList.remove('collection-link-hidden');
         }
-        for (var i = 0; i < inbetweenLinksLength; i++) {
-          inbetweenLinks[0].className = 'collection-link-hidden'
+        for (var k = 0; k < inbetweenLinksLength; k += 1) {
+          inbetweenLinks[0].className = 'collection-link-hidden';
         }
-      }
-    }  
+      };
+    }
   }
-}
\ No newline at end of file
+}
+
+function requestReactionCounts(articleId) {
+  var ajaxReq;
+  if (window.XMLHttpRequest) {
+    ajaxReq = new XMLHttpRequest();
+  } else {
+    ajaxReq = new ActiveXObject('Microsoft.XMLHTTP');
+  }
+  ajaxReq.onreadystatechange = () => {
+    if (ajaxReq.readyState === XMLHttpRequest.DONE) {
+      var json = JSON.parse(ajaxReq.response);
+      json.article_reaction_counts.forEach(reaction => {
+        setReactionCount(reaction.category, reaction.count);
+      });
+      json.reactions.forEach(reaction => {
+        if (document.getElementById('reaction-butt-' + reaction.category)) {
+          showUserReaction(reaction.category, 'not-user-animated');
+        }
+      });
+    }
+  };
+  ajaxReq.open('GET', '/reactions?article_id=' + articleId, true);
+  ajaxReq.send();
+}
+
+function jumpToComments() {
+  document.getElementById('jump-to-comments').onclick = e => {
+    e.preventDefault();
+    document.getElementById('comments').scrollIntoView({
+      behavior: 'instant',
+      block: 'start',
+    });
+  };
+}
+
+function initializeArticleReactions() {
+  setCollectionFunctionality();
+  setTimeout(() => {
+    var articleId;
+    if (document.getElementById('article-body')) {
+      articleId = document.getElementById('article-body').dataset.articleId;
+      if (document.getElementById('article-reaction-actions')) {
+        requestReactionCounts(articleId);
+      }
+    }
+    var reactionButts = document.getElementsByClassName(
+      'article-reaction-butt',
+    );
+    for (var i = 0; i < reactionButts.length; i += 1) {
+      reactionButts[i].onclick = e => {
+        reactToArticle(articleId, this.dataset.category);
```

Comment:
`this.dataset` doesn't exist here, I think it's because you switched from a regular function to an arrow function

Reply:
Oh, I should have been more thorough with my testing! My bad >.< noticed the clicks not registering and throwing console error for the same. I will go through all the functionality once more and confirm nothing else is broken.

Reply:
Checked out all the reaction functionality with the admin panel and respective interfaces! It should be working now.